dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1430,6 +1432,42 @@ void TeachingTip::StartContractToClose()
     }
 }
 
+winrt::TeachingTipPlacementMode TeachingTip::GetPreferredPlacement()
+{
+    const auto isRightToLeft = FlowDirection() == winrt::FlowDirection::RightToLeft;
+    switch (auto&& placementMode = PreferredPlacement())
+    {
+    case winrt::TeachingTipPlacementMode::Top:
```

Comment:
This would be much better as a table-based approach. "GetFlippedPlacement" which indexes into a constant data array of the answers.

Reply:
Agreed.